apis/placement/v1beta1/override_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",categories={fleet,fleet-placement}
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Override defines a group of override policies about how to override the selected resources and how to apply these resources
+// to the target clusters.
+// Note: override may fail, and it will be reflected on the placement status.
+type Override struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ClusterResourcePlacement.
+	// +required
+	Spec OverrideSpec `json:"spec"`
+}
+
+// OverrideSpec defines the desired state of the Override.
+type OverrideSpec struct {
```

Comment:
Better to make overridepolicy part of a placement to avoid creating N-M mappings between overrideds & placements.

You can think a placement as an application. Usually overrides are part of the application configuration.

They need to managed by the same developer. As an owner of the app, I definitely don't want others to arbitrarily override my app.

This is also how kubefed does it.

Reply:
Something like:

```
PlacementSpec {
  ...
  OverridePolicies []OverridePolicy 
}

OverridePolicy {
    // Target clusters.
    ClusterNames []string
    (or) ClusterSelector

    // Target resources.
    ClusterResourceSelectors []ClusterResourceSelector
    ResourceSelectors []ResourceSelector

    // Overrides.
    JSONPatchOverrides []JSONPatchOverride
  }
```

Reply:
I have discussed this option with Ryan and Patrick. We take the current option because of:

- there is no relationship between CRP and override. The override is applying to the resources(s).
- the apply result is determined by the resource itself. The value won't be changed if multiple CRPs select the same resource but use different override policies.